distributed/tests/test_client.py

```diff
@@ -4072,8 +4076,6 @@ async def test_as_current(c, s, a, b):
 
         with temp_default_client(c):
             assert Client.current() is c
-            with pytest.raises(ValueError):
-                Client.current(allow_global=False)
             with c1.as_current():
```

Comment:
In tests I was playing with removing `as_current`.  Presumably `with client.as_current` should now be equivalent to `with client` and shouldn't be something that we recommend any longer.

Reply:
Interesting problem. When I did this I encountered the `Client._close` not awaited problem you mentioned before.

This is introduced (in my case) because I switched `temp_default_client` to use `with client` instead of `with client.as_current()`.

Now, whenever this temp_default_client exits the context, it calls `__exit__` even on asynchronous clients.

https://github.com/dask/distributed/blob/a9c240156544c99a58e6a6925ae34db7ffbc1096/distributed/client.py#L1709-L1713

here we are creating the close task but are never awaiting them.

TLDR `with client` and `with client.as_current()` is not equivalent since `with client` is closing the client as well.

Reply:
Interesting problem. When I did this I encountered the `Client._close` not awaited problem you mentioned before.

This is introduced (in my case) because I switched `temp_default_client` to use `with client` instead of `with client.as_current()`.

Now, whenever this temp_default_client exits the context, it calls `__exit__` even on asynchronous clients.

https://github.com/dask/distributed/blob/a9c240156544c99a58e6a6925ae34db7ffbc1096/distributed/client.py#L1709-L1713

here we are creating the close task but are never awaiting them.

TLDR `with client` and `with client.as_current()` is not equivalent since `with client` is closing the client as well.